src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
 public class SalaryInfo {
+    private static final int POSITION_OF_DATE = 0;
+    private static final int POSITION_OF_NAME = 1;
+    private static final int POSITION_OF_WORKED_HOURS = 2;
+    private static final int POSITION_OF_SALARY_PER_HOURS = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] dataSplit;
+        String[] dataOutput = new String[names.length];
+        int j = 0;
+        for (String name : names) {
+            int sum = 0;
+            for (String oneData : data) {
+                dataSplit = oneData.split(" ");
+                if (dataSplit[POSITION_OF_NAME].equals(name)) {
+                    if (getDate(dataSplit[POSITION_OF_DATE]).getTime()
+                            >= getDate(dateFrom).getTime()) {
+                        if (getDate(dataSplit[POSITION_OF_DATE]).getTime()
+                                <= getDate(dateTo).getTime()) {
```

Comment:
think how to refactor this , you can create a variable, cause more than 1  time you're counting `dataSplit[POSITION_OF_DATE]` or